index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
+   *
+   * @returns Promise of the active public key
+   */
+  getActivePublicKey(): Promise<string>;
+
+  /**
+   * Add a token to ArConnect (if it is not already present)
+   *
+   * @param id Token contract ID
+   * @param type Type of the token (asset or collectible)
+   * @param gateway Gateway config for the token
+   */
+  addToken(id: string, type?: TokenType, gateway?: GatewayConfig): Promise<void>;
+
+  /**
+   * Checks if a token has been added to ArConnect
+   * 
+   * @param id Token to check for
+   * 
+   * @returns Token added or not
+   */
+  isTokenAdded(id: string): Promise<boolean>;
+
+  /**
+   * Dispatch an Arweave transaction (preferably bundled)
+   *
+   * @param transaction Transaction to dispatch
+   * 
+   * @returns Dispatched transaction ID and type
+   */
+  dispatch(transaction: Transaction): Promise<DispatchResult>;
+
+  /**
+   * Create a deterministic secret based on the input data.
+   * 
+   * @param data Input data to generate the hash from
+   * @param options Hash configuration
+   * 
+   * @returns Hash array
+   */
+  privateHash(data: ArrayBuffer, options: SignMessageOptions): Promise<Uint8Array>; // TODO: data is ArrayBuffer or BinaryDataType
+
+  /**
+   * Create and sign a DataItem (bundled transaction item),
+   * that can be loaded into "arbundles".
+   * 
+   * @param dataItem Data item params
+   * 
+   * @returns Buffer of the signed data item
+   */
+  signDataItem(dataItem: DataItem): Promise<ArrayBufferLike>; // TODO: Is this ArrayBufferLike or Buffer based on the example? `const dataItem = new DataItem(Buffer.from(result));` The implementation says `return Array.from<number>(dataEntry.getRaw());` in https://github.com/arconnectio/ArConnect/blob/production/src/api/modules/sign_data_item/sign_data_item.background.ts
+
+  /**
+   * Create a cryptographic signature for any piece of data for later validation.
+   * This function cannot be used to sign transactions or interactions, because the data
+   * gets hashed before the signature generation.
+   * 
+   * @param data Message to be hashed and signed
+   * @param options Signature options
+   * 
+   * @returns Signature
+   */
+  signMessage(data: ArrayBuffer, options?: SignMessageOptions): Promise<Uint8Array>; // TODO: data is ArrayBuffer or BinaryDataType
+
+  /**
+   * Verify a cryptographic signature created with the arweaveWallet.signMessage() function.
+   * 
+   * @param data Data to verify with the signature
+   * @param signature Signature to validate
+   * @param publicKey Optionally match the signature with a different public key than the currently active
+   * @param options Signature options
+   * 
+   * @returns Validity
+   */
+  verifyMessage(
+    data: ArrayBuffer, // TODO: data is ArrayBuffer or BinaryDataType
+    signature: ArrayBuffer | string, // TODO: signature is ArrayBuffer or BinaryDataType
+    publicKey?: string,
+    options?: SignMessageOptions
+  ): Promise<boolean>;
```

Comment:
Similar to `privateHas() and `signMessage()`. See above.

Also, if `signMessage()` returns `Uint8Array`, that cannot be passed to `verifyMessage()` without conversion, as shown in the docs:

![image](https://github.com/user-attachments/assets/859b2f05-b488-47b4-b7b9-2f86b1c0b6f3)